Node-1st-gen/fulltext-search-firestore/functions/index.js

```diff
@@ -14,24 +14,25 @@
  * limitations under the License.
  */
 const functions = require('firebase-functions/v1');
+const {defineSecret} = require('firebase-functions/params');
 const algoliasearch = require('algoliasearch').default;
 
 // [START init_algolia]
 // Initialize Algolia, requires installing Algolia dependencies:
 // https://www.algolia.com/doc/api-client/javascript/getting-started/#install
 //
 // App ID and API Key are stored in functions config variables
-const ALGOLIA_ID = functions.config().algolia.app_id;
-const ALGOLIA_ADMIN_KEY = functions.config().algolia.api_key;
-const ALGOLIA_SEARCH_KEY = functions.config().algolia.search_key;
+const ALGOLIA_ID = defineSecret('ALGOLIA_ID');
+const ALGOLIA_ADMIN_KEY = defineSecret('ALGOLIA_ADMIN_KEY');
+const ALGOLIA_SEARCH_KEY = defineSecret('ALGOLIA_SEARCH_KEY');
 
 const ALGOLIA_INDEX_NAME = 'notes';
-const client = algoliasearch(ALGOLIA_ID, ALGOLIA_ADMIN_KEY);
 // [END init_algolia]
 
 // [START update_index_function]
 // Update the search index every time a blog post is written.
-exports.onNoteCreated = functions.firestore.document('notes/{noteId}').onCreate((snap, context) => {
+exports.onNoteCreated = functions.runWith({secrets: ["ALGOLIA_ID", "ALGOLIA_ADMIN_KEY"]}).firestore.document('notes/{noteId}').onCreate((snap, context) => {
+  const client = algoliasearch(ALGOLIA_ID.value(), ALGOLIA_ADMIN_KEY.value());
```

Comment:
use one global instance of this client instead of initializing each time the function runs. In order to access param values, you need to do this inside of an onInit hook https://firebase.google.com/docs/functions/config-env#init-fn